pytorch_lightning/plugins/environments/lsf_environment.py

```diff
@@ -52,51 +55,128 @@ def __init__(self) -> None:
             )
         self._main_address = self._get_main_address()
         self._main_port = self._get_main_port()
-        log.debug(f"MASTER_ADDR: {self._main_address}")
-        log.debug(f"MASTER_PORT: {self._main_port}")
+        self._local_rank = self._get_local_rank()
```

Comment:
Are there concerns about these instance variables going stale vs the contents of the file/env? Why not always look these up from the env instead of having these cached values?

Reply:
> Are there concerns about these instance variables going stale vs the contents of the file/env? 

No

> Why not always look these up from the env instead of having these cached values?

I guess it's just a coding preference of mine. I'm not completely familiar with how these values get used, but if they're accessed more than once, reading from the environment at every look up seems inefficient. For example, setting `main_address` and `node_rank` require opening and reading a file. The performance impact is probably within the noise in this context, but, like I said, this is just a stylistic preference of mine.

Reply:
You can also use

```py
from functools import lru_cache

@property
@lru_cache(1)
def local_rank(self):
    ...
```

and avoid the `_get_*` methods

Reply:
world size, global rank, local rank and address come from the environment variables and can be looked up in constant time. I would prefer to keep it this way, but not a big request I'm also fine with this.